### PR TITLE
changed cert checker to use ruby openssl to generate sha1 fingerprint

### PR DESCRIPTION
**Pull request by Cyberbeni**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

Changed the used openssl in cert_checker to generate sha1 fingerprint from the one used in terminal to the ruby version that's used in most places, because that's the one printing the version number in fastlane env.
https://github.com/fastlane/fastlane/pull/11641

https://stackoverflow.com/a/16529867/6324550

### Description

Using ruby openssl instead of terminal one in cert_checker.

### CODE REVIEW

**Comment by KrauseFx** [comment]
Awesome, great PR, thanks for your contribution @Cyberbeni. Did you test this change locally and verify things work as expected?

**Comment by KrauseFx** [comment]
Test 😢 
```
SyntaxError: /Users/distiller/project/fastlane_core/lib/fastlane_core/cert_checker.rb:115: syntax error, unexpected keyword_end, expecting end-of-input
  /Users/distiller/project/fastlane_core/lib/fastlane_core.rb:14:in `require_relative'
  /Users/distiller/project/fastlane_core/lib/fastlane_core.rb:14:in `<top (required)>'
  /Users/distiller/project/fastlane/lib/fastlane.rb:1:in `require'
```

**Comment by Cyberbeni** [comment]
Hmm... Awkward, I thought I'd be safe with just running `bundle exec rspec` and `bundle exec rubocop -a` on this one.  Did a quick check and it wasn't giving the expected result, so I fixed it, now self.sha1_fingerprint gives the same result as before.

File: fastlane_core/lib/fastlane_core/cert_checker.rb [hunk 1]
```diff
@@ -102,15 +103,12 @@ def self.wwdr_keychain
...
-        return result
-      rescue
-        UI.message(result)
-        UI.user_error!("Error parsing certificate '#{path}'")
-      end
+      file_data = File.read(path.to_s)
+      cert = OpenSSL::X509::Certificate.new(file_data)
+      return OpenSSL::Digest::SHA1.new(cert.to_der).to_s.upcase
+    rescue => error
+      UI.message(error)
```
Comment (lacostej) [inline fastlane_core/lib/fastlane_core/cert_checker.rb]: maybe UI.error?

**Comment by janpio** [comment]
@Cyberbeni Could you maybe rebase this onto current master? Then I can also run the Windows tests on this manually...

**Review by janpio — APPROVED** [review]

**Comment by janpio** [comment]
🥇

**Pull request merged**